libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2663,9 +2663,16 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 		))
 		{
 			annotation.isPure = true;
-			m_scope->annotation().contractDependencies.insert(
-				&dynamic_cast<ContractType const&>(*magicType->typeArgument()).contractDefinition()
-			);
+			ContractType const& accessedContractType = dynamic_cast<ContractType const&>(*magicType->typeArgument());
+			m_scope->annotation().contractDependencies.insert(&accessedContractType.contractDefinition());
+
+			if (!accessedContractType.immutableVariables().empty())
+				m_errorReporter.typeError(
+					9274_error,
+					_memberAccess.location(),
+					"\"runtimeCode\" is not available for contracts containing immutable variables."
```

Comment:
Wait a second - this will be triggered for ``creationCode`` as well now, won't it?
``creationCode`` is perfectly fine in the presence of immutables, so this shouldn't be an error - and we should best have a test for it.

Reply:
You are absolutely right. The amount of oversights in this PR are increasing and I am feeling a bit embarrassed about that